File: Data Structures & Algorithms/products-of-array-discluding-self/submission-0.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        ilocz = 1
        ans = list([])

        if 0 in nums:
            if nums.count(0) > 1:
                for i in nums:
                    ans.append(0)
            else:
                for i in nums:
                    if i != 0:
                        ilocz*=i
                for i in nums:
                    if i == 0:
                        ans.append(ilocz)
                    else:
                        ans.append(0)         
            return ans
        else:
            for i in nums:
                ilocz *= i

            for i in nums:
                ans.append(int(ilocz/i))
            
            return ans

File: Data Structures & Algorithms/permutations/submission-0.cpp
class Solution {
public:
   void dfs(vector<int>& nums, vector<int>&ans, vector<vector<int>>& endgame, vector<bool>& bv)
   {
        if(ans.size() == nums.size())
        {
             endgame.push_back(ans);
             return;
        }
        for(int i =0; i < nums.size(); i++)
        {
            if(bv[i] == false)
            {
                ans.push_back(nums[i]);
                bv[i] = true;
                dfs(nums, ans, endgame, bv);
                bv[i] = false;
                ans.pop_back();
            }
        }
        
   }

    vector<vector<int>> permute(vector<int>& nums) {
        vector<bool>bv(nums.size(), false);
        vector<int> ans;
        vector<vector<int>>endgame;

        dfs(nums, ans, endgame, bv);
        return endgame;
    }
};
